### PR TITLE
starlark: permit if/for/while at toplevel if -globalreassign

### PR DESCRIPTION
**Pull request by alandonovan**
Also:
- make the option handling work the same across all tests
- be explicit about which options are needed in each test chunk
- improve documentation of options
- rename fp option to float consistently
- rename global_reassign option to globalreassign consistently

### CODE REVIEW

**Comment by glycerine** [comment]
It could be my merge mistake, but its worth checking at the top level that `while` will work, even with resolve.AllowGlobalReassign = true
~~~
n=6
while n > 0:
   n=n-1
   print(n)

~~~
I see endings 5s:
~~~
5
5
5
5
5
....
~~~

**Comment by glycerine** [comment]
possibly related
~~~
m=6 # at toplevel
def g():
  while m>0:
    m=m-1
    print(m)

~~~
gives
~~~
>>> g()
Traceback (most recent call last):
  <stdin>:1: in <expr>
  <stdin>:2: in g
Error: local variable m referenced before assignment
~~~

**Comment by alandonovan** [comment]
Works for me at this commit:
```
$ git co options
Already on 'options'
Your branch is up to date with 'origin/options'.
$ go build -o x ./cmd/starlark
$ cat a.go
n=6
while n > 0:
   n=n-1
   print(n)

$ ./x -recursion -globalreassign a.go
5
4
3
2
1
0
```
If you can reproduce the problem at this or any other commit, please do file a bug. 

Regarding "local variable m referenced before assignment": yes, that's the intended behavior. Starlark does not have Python's global (or nonlocal) keyword so you just have to choose different variable names. The global and nonlocal statements would be easy enough to implement, but given the easy workaround, there's little urgency.

File: doc/spec.md [hunk 1]
```diff
@@ -254,14 +252,11 @@ appear in the grammar; they are reserved as possible future keywords:
...
-from           with
-global         yield
+as             finally        nonlocal
+assert         from           raise
+class          global         try
+del            import         with
+except         is             yield
 ```
 
 <b>Implementation note:</b>
```
Comment (jayconrod) [inline doc/spec.md]: Consider adding implementation note for `while`, since it's not a keyword in the default grammar.

**Pull request merged**